### PR TITLE
draw text stuff, online, etc.

### PR DESCRIPTION
**Pull request by Mr-Auto**
- hide Vec2 constructor that is not valid in lua
- fix and expose some online stuff
- Add constructor for `TextRenderingInfo` and `draw_text` that takes it as parameter as optimization and future development
- replace `prepare_text_for_rendering` with `set_text` method on  `TextRenderingInfo`
- add function to change and rotate text in `TextRenderingInfo` with necessary structs etc.

### CODE REVIEW

File: src/game_api/script/usertypes/vanilla_render_lua.cpp [hunk 1]
```diff
@@ -285,6 +294,7 @@ void register_usertypes(sol::state& lua)
 
     lua.new_usertype<TextRenderingInfo>(
         "TextRenderingInfo",
+        // sol::constructors<TextRenderingInfo(const std::string, float, float, float, float, uint32_t, uint32_t)>{},
```
Comment (Mr-Auto) [inline src/game_api/script/usertypes/vanilla_render_lua.cpp:297]: not sure what I'm doing wrong, but this just insta crash when you try to constructs new one

File: src/game_api/render_api.cpp [hunk 2]
```diff
@@ -665,3 +686,51 @@ TextRenderingInfo::~TextRenderingInfo()
         game_free(unknown6);
     }
 }
+void TextRenderingInfo::rotate(float angle, std::optional<float> px, std::optional<float> py)
+{
+    const auto resx = static_cast<float>(get_setting(GAME_SETTING::RESOLUTIONX).value_or(16));
+    const auto resy = static_cast<float>(get_setting(GAME_SETTING::RESOLUTIONY).value_or(9));
```
Comment (Dregu) [inline src/game_api/render_api.cpp:692]: These just return my ultrawide fullscreen resolution, while I'm in a tiny window no less...

Comment (Mr-Auto) [reply src/game_api/render_api.cpp:692]: oh, is there a good way of getting the screen ratio then? i know the game forces 16:9 but still fills strange to hardcode those values

Comment (Dregu) [reply src/game_api/render_api.cpp:692]: 16/9 was written in so many places I had to abandon the ultrawide mod

File: docs/src/includes/_types.md [hunk 3]
```diff
@@ -3307,30 +3307,30 @@ Type | Name | Description
...
-float | [x](https://github.com/spelunky-fyi/overlunky/search?l=Lua&q=x) | 
-float | [y](https://github.com/spelunky-fyi/overlunky/search?l=Lua&q=y) | 
-int | [layer](https://github.com/spelunky-fyi/overlunky/search?l=Lua&q=layer) | 
-float | [width](https://github.com/spelunky-fyi/overlunky/search?l=Lua&q=width) | 
-float | [height](https://github.com/spelunky-fyi/overlunky/search?l=Lua&q=height) | 
-float | [special_offsetx](https://github.com/spelunky-fyi/overlunky/search?l=Lua&q=special_offsetx) | 
-float | [special_offsety](https://github.com/spelunky-fyi/overlunky/search?l=Lua&q=special_offsety) | 
-float | [tile_width](https://github.com/spelunky-fyi/overlunky/search?l=Lua&q=tile_width) | 
-float | [tile_height](https://github.com/spelunky-fyi/overlunky/search?l=Lua&q=tile_height) | 
+int | [flags](https://github.com/spelunky-fyi/overlunky/search?l=Lua&q=flags) | see [flags.hpp](https://github.com/spelunky-fyi/overlunky/blob/main/src/game_api/flags.hpp) entity_flags
```
Comment (Dregu) [inline docs/src/includes/_types.md:3312]: Should probably set the type of these to ENT_FLAG etc, that's more useful than flags.hpp.

File: docs/src/includes/_types.md [hunk 4]
```diff
@@ -3307,30 +3307,30 @@ Type | Name | Description
...
+float | [special_offsety](https://github.com/spelunky-fyi/overlunky/search?l=Lua&q=special_offsety) | Special offset used for entities attached to others (or picked by others) that need to flip to the other side when the parent flips sides
+float | [tile_width](https://github.com/spelunky-fyi/overlunky/search?l=Lua&q=tile_width) | Size of the sprite in the texture
+float | [tile_height](https://github.com/spelunky-fyi/overlunky/search?l=Lua&q=tile_height) | Size of the sprite in the texture
 float | [angle](https://github.com/spelunky-fyi/overlunky/search?l=Lua&q=angle) | 
 [Color](#Color) | [color](https://github.com/spelunky-fyi/overlunky/search?l=Lua&q=color) | 
-float | [hitboxx](https://github.com/spelunky-fyi/overlunky/search?l=Lua&q=hitboxx) | 
-float | [hitboxy](https://github.com/spelunky-fyi/overlunky/search?l=Lua&q=hitboxy) | 
+float | [hitboxx](https://github.com/spelunky-fyi/overlunky/search?l=Lua&q=hitboxx) | Width of the hitbox
+float | [hitboxy](https://github.com/spelunky-fyi/overlunky/search?l=Lua&q=hitboxy) | Height of the hitbox
```
Comment (Dregu) [inline docs/src/includes/_types.md:3329]: Half actually

File: docs/src/includes/_types.md (hunk 3 above)
Comment (Mr-Auto) [reply docs/src/includes/_types.md]: Sure, but I'll leave the comment

**Pull request merged**